### ISSUE TITLE
YouTube or Vimeo embed with autoplay?

### ISSUE THREAD

**Issue by proimage** [wrav/oembed#14]
I can't for the life of me figure out how to use this oEmbed plugin to parse a URL to a YouTube or Vimeo video and render an iframe with control over things like width/height and autoplay. Could you possibly provide a code example?

The problem I'm running into seems to be that the `options` object does nothing to affect the output. Here's my code:

```
{% set url = entry.oEmbedField %}
{% set options = {
	autoplay: 1,
	width: "800px",
	height: "450px"
 } %}
{{ craft.oembed.render(url, options) }}
```

The output ends up being this (linebreaks added for readability):

```
<iframe
    width="480"
    height="270"
    src="https://www.youtube.com/embed/[VIDEO_ID_HERE]?feature=oembed"
    frameborder="0"
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowfullscreen=""></iframe>
```

Thanks for any help you can provide!

**Comment by reganlawton** [comment]
I’m almost certain this is a feature you add in the URL, something like <URL>?autoplay=1. I’ll have to check when I’m at my computer.

**Comment by gudehus** [comment]
I have the same problem with Vimeo videos. But they have a documentation that listed all the options.
https://developer.vimeo.com/api/oembed/videos

Maybe this is helping. And some code example would be helpful.

**Comment by reganlawton** [comment]
It seems the library `embed` is stripping the URL I'll have to have a play with this.

**Comment by reganlawton** [comment]
I think I have a way around this by using PHP to edit the DOM on the iframe. I'll need to look into this abit more.

**Comment by reganlawton** [comment]
Seems to be linked to this [issue](https://github.com/oscarotero/Embed/issues/263), I still think DOM manipulation might be the way I've still got some playing around to do

**Comment by reganlawton** [comment]
@proimage and @gudehus are you able to checkout using composer, `dev-feature/autoplay`. It should let you support autoplay on videos. Its hijacking the URLs and reformatting it. I also added more support for additional vimeo options.

**Comment by reganlawton** [comment]
Deployed fix in v1.1.8

**Comment by linehammer** [comment]
Since April 2018, Google made some changes to the  Autoplay Policy. You not only need to add the autoplay=1 as a query param, but also add allow='autoplay' as an iframe's attribute

So while [embed youtube](http://net-informations.com/q/mis/youtube.html) video, you will have to do something like this:

`<iframe src="https://www.youtube.com/embed/VIDEO_ID?autoplay=1" allow='autoplay'></iframe>`
